src/directory/generateDirectory.mjs

```diff
@@ -0,0 +1,97 @@
+import { promises as fs } from 'fs';
+import { fileURLToPath } from 'url';
+import { cwd } from 'node:process';
+import path from 'path';
+import JSON5 from 'json5';
+import { directory } from './directory.mjs';
+import { writeFile } from 'fs/promises';
+
+/**
+ * Helper function to use RegEx to grab the "meta" object
+ * @param {string} filePath
+ * @returns
+ */
+async function getMetaStringObj(filePath) {
+  const regex = /const\s+meta\s*=\s*(\{[^}]+\})/;
+
+  const file = await fs.readFile(filePath, 'utf-8');
+
+  const match = file.match(regex);
+
+  if (match && match[1]) {
+    try {
+      // Using JSON5 because the meta object is a "relaxed" JSON
+      // JSON5 can parse the meta object without needing quotes around the object keys
+      const result = JSON5.parse(match[1]);
+
+      return result;
+    } catch (err) {
+      throw new error(`Unable to parse meta object. Error:, ${err}`);
+    }
+  }
+}
+const rootPath = path.resolve(cwd(), 'src/pages');
+
+/**
+ * Traverses the directoryNode parameter and updates itself with properties
+ * found from the files
+ * @param {import('./directory').PageNode} directoryNode
+ */
+async function traverseDirectoryObject(directoryNode) {
+  if (directoryNode) {
+    if (directoryNode.isExternal) {
+      // no op
+    } else if (directoryNode.path) {
+      const metaObj = await getMetaStringObj(directoryNode.path);
+
+      if (metaObj) {
+        for (const key of Object.keys(metaObj)) {
+          directoryNode[key] = metaObj[key];
+        }
```

Comment:
Here I'm just grabbing all the properties of the `meta` object. I could define an object in this file so that we only grab specific properties, let me know what you think